src/PRAS2PowerSystems.jl

```diff
@@ -1,240 +1,265 @@
-            PRASCore.StorageAvailability(),
-            PRASCore.GeneratorStorageAvailability(),
-            PRASCore.LineAvailability(),
-        )
-    @info "Successfully completed PRAS Runs. Exporting outage profiles now..."
-    # Setup to save the new PSY System with scenario timeseries data
-    #working_dir = pwd();
-    working_dir = location
-    dt_now = Dates.format(Dates.now(), "dd-u-yy-H-M-S")
-
-    dir_name = joinpath(
-        working_dir,
-        "data",
-        "Generated-Outage-Profile-JSON",
-        string(UUIDs.uuid4()),
-        dt_now,
-    )
+Add the asset status from the worst sample to PSY.TimeSeriesForcedOutage of the component.
+
+# Arguments
+
+  - `sys::PSY.System`: PowerSystems.jl system model
+  - `results::T`: Tuple of results from the PRAS assess call
+  - `template::RATemplate`: PRAS problem template
 
-    #  **TODO: IF generating systems for multiple scenarios, remember to delete the timeseries data from previous scenario.
-    asset_dict = Dict([
-        (:generators, (gens_avail, PSY.Generator)),
-        (:storages, (stors_avail, PSY.Storage)),
-        (:generatorstorages, (gen_stors_avail, PSY.StaticInjection)),
-        (:lines, (lines_avail, PSY.Branch)),
-    ])
-    asset_keys = []
-
-    for k in keys(asset_dict)
-        if (length(getfield(pras_system, k)) != 0)
-            push!(asset_keys, k)
+# Returns
+
+    - PSY System with PSY.TimeSeriesForcedOutage for all components for which asset status is available
+"""
+function add_asset_status!(
+    sys::PSY.System,
+    results::T,
+    template::RATemplate,
+) where {T <: Tuple{Vararg{PRASCore.Results.Result}}}
+    @info ""
+    # Time series timestamps
+    filter_func = x -> (typeof(x) <: PSY.StaticTimeSeries)
+    all_ts = PSY.get_time_series_multiple(sys, filter_func)
+    ts_timestamps = TimeSeries.timestamp(first(all_ts).data)
+
+    shortfall_samp_idx =
+        findfirst(name.(typeof.(results)) .== PRASCore.Results.ShortfallSamplesResult)
+    shortfall_samples = results[shortfall_samp_idx]
+
+    sample_idx = sortperm(shortfall_samples[], rev=true)[1]
+    resultsspecs = get_outage_pras_resultspec(template)
+
+    for result_spec in filter(x -> !(x == ShortfallSamples()), resultsspecs)
+        device_ramodel = get_device_ramodel(typeof(result_spec))
+        gens_to_formula =
+            build_component_to_formulation(device_ramodel, sys, template.device_models)
+
+        gens_avail_idx =
+            findfirst(name.(typeof.(results)) .== get_pras_resulttype(typeof(result_spec)))
+        gens_avail = results[gens_avail_idx]
+
+        for gen in keys(gens_to_formula)
+            ts_forced_outage =
+                PSY.TimeSeriesForcedOutage(; outage_status_scenario="WorstShortfallSample")
+            PSY.add_supplemental_attribute!(sys, gen, ts_forced_outage)
+
+            availability_data = TimeSeries.TimeArray(
+                ts_timestamps,
+                getindex.(gens_avail[gen.name, :], sample_idx),
```

Comment:
This is where we actually use the relationship between names in PRAS and Sienna, if I understand it correctly?

Reply:
Yes, this is correct.